server/configuration.go

```diff
@@ -165,6 +165,9 @@ func (p *Plugin) OnConfigurationChange() error {
 	}
 
 	if err := p.validateConfiguration(configuration); err != nil {
+		if p.configuration == nil {
+			p.setConfiguration(configuration)
+		}
```

Comment:
The `getConfiguration` handles this -- within the mutex -- by returning `&configuration{}`. Is there a reason we need to deal with `configuration == nil` again?